subsys/net/ip/net_core.c

```diff
@@ -65,12 +66,66 @@ void net_analyze_stack(const char *name, const char *stack, size_t size)
 }
 #endif /* CONFIG_INIT_STACKS */
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+static inline void process_raw_data(struct net_pkt *pkt)
+{
+	struct net_context *cont;
+	struct net_pkt *raw_pkt;
+	int ret;
+
+	cont = net_context_find(AF_PACKET, SOCK_RAW, ETH_P_ALL);
+	if (!cont) {
+		return;
+	}
+
+	raw_pkt = net_pkt_clone(pkt, K_MSEC(100));
+	if (!raw_pkt) {
+		return;
+	}
+
+	net_pkt_set_family(raw_pkt, AF_PACKET);
```

Comment:
@tbursztyka Important issue is here. When core receives packet from driver, core will call L2 to verify header and remove L2 header. In case of raw socket, we have to just call L2 to verify and not to exclude.
Second issue is, If user opens a raw socket and non-raw socket. Then same packet has to be feed to raw socket with L2 header and for non-raw socket only data has to be feed.
Do we need to clone or something else? any suggestions?